### PR TITLE
[FSDE-31479] Implement autocomplete mode in the search input Chi documentation

### PR DESCRIPTION
**Pull request by pedrotaglialatela-lumen**
https://ctl.atlassian.net/browse/FSDE-31479

### CODE REVIEW

File: src/website/views/components/forms/search-input/_examples.pug [hunk 1]
```diff
@@ -116,19 +116,17 @@ p.-text
...
       .chi-col.-w--12.-w-sm--8.-w-md--6.-w-lg--5
         .chi-form__item
-          chi-search-input#example__with_dropdown
-          chi-dropdown#example__dropdown(prevent-auto-hide position='bottom')
-            a.chi-dropdown__menu-item(href='#' slot='menu') Item 1
-            a.chi-dropdown__menu-item(href='#' slot='menu') Item 2
-            a.chi-dropdown__menu-item(href='#' slot='menu') Item 3
-            a.chi-dropdown__menu-item(href='#' slot='menu') Item 4
-
+          chi-search-input#example_autocomplete(mode="autocomplete")
```
Comment (josecarlosgonzalezv) [inline src/website/views/components/forms/search-input/_examples.pug:129]:
```suggestion
          chi-search-input#example__autocomplete(mode="autocomplete")
```

File: src/website/views/components/forms/search-input/_examples.pug [hunk 2]
```diff
@@ -149,26 +147,19 @@ h3 With Dropdown
...
     :code(lang="html")
       <div class="chi-form__item">
-        <chi-search-input id="example__with_dropdown"></chi-search-input>
-        <chi-dropdown id="example__dropdown" position="bottom" prevent-auto-hide>
-          <a class="chi-dropdown__menu-item" href="#" slot="menu">Item 1</a>
-          <a class="chi-dropdown__menu-item" href="#" slot="menu">Item 2</a>
-          <a class="chi-dropdown__menu-item" href="#" slot="menu">Item 3</a>
-          <a class="chi-dropdown__menu-item" href="#" slot="menu">Item 4</a>
-        </chi-dropdown>
+        <chi-search-input id="example_autocomplete" mode="autocomplete"></chi-search-input>
```
Comment (josecarlosgonzalezv) [inline src/website/views/components/forms/search-input/_examples.pug:150]:
```suggestion
        <chi-search-input id="example__autocomplete" mode="autocomplete"></chi-search-input>
```

File: src/website/views/components/forms/search-input/_examples.pug [hunk 3]
```diff
@@ -149,26 +147,19 @@ h3 With Dropdown
...
       </div>
 
       <script>
-        const input = document.getElementById("example__with_dropdown");
-        const dropdown = document.getElementById("example__dropdown");
-
-        input.addEventListener("chiFocus", () => {
-          dropdown.show();
-        });
+        const input = document.getElementById("example_autocomplete");
```
Comment (josecarlosgonzalezv) [inline src/website/views/components/forms/search-input/_examples.pug:154]:
```suggestion
        const input = document.getElementById("example__autocomplete");
```

**Review by josecarlosgonzalezv — APPROVED** [review]

File: src/custom-elements/src/components/search-input/search-input.scss [hunk 4]
```diff
@@ -1,6 +1,10 @@
...
-    @import '../../global/styles/common';
+  @import '../../global/styles/common';
 
-    display: block;
-    width: 100%;
+  display: flex;
+
+  &[mode="autocomplete"] {
+    flex-direction: column-reverse;
+    position: relative;
```
Comment (jllr) [inline src/custom-elements/src/components/search-input/search-input.scss:8]:
```suggestion
  display: block;
  width: 100%;

  &[mode="autocomplete"] {
    display: flex;
    flex-direction: column-reverse;
    position: relative;
```

Comment (jllr) [reply src/custom-elements/src/components/search-input/search-input.scss:8]: @pedrotaglialatela-lumen doing so we make sure it works in both cases

**Review by josecarlosgonzalezv — APPROVED** [review]

**Review by jllr — APPROVED** [review]

**Pull request merged**